### PR TITLE
Added function to get the all jobs in the python sdk client

### PR DESCRIPTION
**Pull request by JohanWork**
Adding functionally to return a list of jobs. Also adding functionality for getting only recurrent jobs . 
Believe it would be good to add this to be able to se jobs without having to jus the ui. The PR also contains some update of reference that was broken. I am still working on this pr but posting to get som feedback and help

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/3227)

### CODE REVIEW

**Comment by JohanWork** [comment]
@IronPan , @hongye-sun  I am trying to filter out the jobs that only are enabled but having a hard time getting my filter right. Do you have any comments or can point me in the right direction?

File: sdk/python/kfp/_client.py [hunk 1]
```diff
@@ -474,6 +474,21 @@ def schedule_pipeline(self, experiment_id, job_name, pipeline_package_path=None,
...
+    """List jobs.
+    Args
+      page_token: token for starting of the page.
+      page_size: size of the page.
+      sort_by: one of 'field_name', 'field_name des'. For example, 'name des'.
+      recurrent: boolean to filter for only recurrent jobs
+    Returns:
+      A response object including a list of jobs and next page token.
+    """
+    if recurrent==True:
```
Comment (parthmishra) [inline sdk/python/kfp/_client.py]:
```suggestion
    if recurrent is True: 
```

File: sdk/python/kfp/_client.py [hunk 2]
```diff
@@ -474,6 +474,21 @@ def schedule_pipeline(self, experiment_id, job_name, pipeline_package_path=None,
...
+      page_size: size of the page.
+      sort_by: one of 'field_name', 'field_name des'. For example, 'name des'.
+      recurrent: boolean to filter for only recurrent jobs
+    Returns:
+      A response object including a list of jobs and next page token.
+    """
+    if recurrent is True: 
+      response = self._job_api.list_jobs(page_token='', page_size=10, sort_by='',resource_reference_key_type=kfp_server_api.models.api_resource_type.ApiResourceType.JOB,filter='')
+    else: 
+      response =self._job_api.list_jobs(page_token='', page_size=10, sort_by='')
```
Comment (NikeNano) [inline sdk/python/kfp/_client.py]:
```suggestion
      response = self._job_api.list_jobs(page_token='', page_size=10, sort_by='')
```

**Comment by NikeNano** [comment]
> @IronPan , @hongye-sun I am trying to filter out the jobs that only are enabled but having a hard time getting my filter right. Do you have any comments or can point me in the right direction?

I believe it needs to be on a row or comment of it's own.

**Comment by JohanWork** [comment]
Kind reminder @hongye-sun

**Pull request closed without merging**